torchvision/transforms/transforms.py

```diff
@@ -683,10 +713,13 @@ def forward(self, img):
         Returns:
             PIL Image or Tensor: Randomly transformed image.
         """
+
+        fillcolor = self._cast_fillcolor(img)
```

Comment:
Let's use a private helper method like `_cast_fillcolor` or `_setup_fillcolor` or another method which we could use to cast fill to float if input is tensor. 

That's true that there is a mismatch in namings: fill or fillcolor depending on the op (https://github.com/pytorch/vision/issues/2479). We have to harmonize that too, but in another PR.

Reply:
Thanks for the reviews!
Let me break down just how tricky the single type thing in TorchScript is here (*it's been bugging me for days*). For instance, if I write a private standalone function _cast_fillcolor(), it must be called from `forward()` since only then would we know the image is tensor, but fillcolor would have to be the input for that function, which won't work since fillcolor is multi-type and anything called from `forward()` is casted in TorchScript thus needed to be single type.
The only way to support a multi-type variable is putting it as a class attribute, as in `self.xxx`. So, the only solution probably is implementing _cast_fillcolor() as a private class method so we can use self.fillcolor, and I use that new base class to avoid writing the same method for all 3 classes.